ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionsQuestionnaireMapper.java

```diff
@@ -0,0 +1,223 @@
+import static uk.gov.hmcts.cmc.ccd.util.StreamUtil.asStream;
+
+@Component
+public class DirectionsQuestionnaireMapper implements Mapper<CCDDirectionsQuestionnaire, DirectionsQuestionnaire> {
+
+    private final ExpertRowMapper expertRowMapper;
+    private final UnavailableDateMapper unavailableDateMapper;
+    private final AddressMapper addressMapper;
+    private final YesNoMapper yesNoMapper;
+
+    @Autowired
+    public DirectionsQuestionnaireMapper(
+        ExpertRowMapper expertRowMapper,
+        UnavailableDateMapper unavailableDateMapper,
+        AddressMapper addressMapper,
+        YesNoMapper yesNoMapper
+    ) {
+        this.expertRowMapper = expertRowMapper;
+        this.unavailableDateMapper = unavailableDateMapper;
+        this.addressMapper = addressMapper;
+        this.yesNoMapper = yesNoMapper;
+    }
+
+    @Override
+    public CCDDirectionsQuestionnaire to(DirectionsQuestionnaire directionsQuestionnaire) {
+        if (directionsQuestionnaire == null) {
+            return null;
+        }
+
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder = CCDDirectionsQuestionnaire.builder();
+        directionsQuestionnaire.getRequireSupport().ifPresent(toRequireSupport(builder));
+
+        toHearingLocation(directionsQuestionnaire.getHearingLocation(), builder);
+
+        directionsQuestionnaire.getWitness().ifPresent(toWitness(builder));
+
+        directionsQuestionnaire.getExpertRequest().ifPresent(toExpertRequest(builder));
+
+        builder.expertReports(directionsQuestionnaire.getExpertReports()
+            .stream()
+            .map(expertRowMapper::to)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList())
+        );
+
+        builder.unavailableDates(directionsQuestionnaire.getUnavailableDates()
+            .stream()
+            .map(unavailableDateMapper::to)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList())
+        );
+
+        return builder.build();
+    }
+
+    private Consumer<RequireSupport> toRequireSupport(
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder
+    ) {
+        return requireSupport -> {
+            requireSupport.getLanguageInterpreter().ifPresent(builder::languageInterpreted);
+            requireSupport.getSignLanguageInterpreter().ifPresent(builder::signLanguageInterpreted);
+            requireSupport.getOtherSupport().ifPresent(builder::otherSupportRequired);
+
+            requireSupport.getHearingLoop()
+                .map(YesNoOption::name)
+                .map(CCDYesNoOption::valueOf)
+                .ifPresent(builder::hearingLoop);
+
+            requireSupport.getDisabledAccess()
+                .map(YesNoOption::name)
+                .map(CCDYesNoOption::valueOf)
+                .ifPresent(builder::disabledAccess);
+        };
+    }
+
+    private Consumer<ExpertRequest> toExpertRequest(
```

Comment:
Consumer should not return, change it to supplier

Reply:
This can not be a supplier.

Reply:
The consumer doesn't return. The consumer is _being_ returned.

Reply:
Whats the reason to return a Consumer. Why not declare it?